--- xserver-xorg-video-psb-0.32.0/src/psb_accel.c	2009-05-11 18:37:53.000000000 -0700
+++ xserver-xorg-video-psb-0.32.0/src/psb_accel.c.new	2010-09-24 06:47:01.221513179 -0700
@@ -501,9 +501,14 @@
 		     PicturePtr pSrcPicture, PicturePtr pMaskPicture,
 		     PicturePtr pDstPicture)
 {
+    int w, h;
     DrawablePtr pDraw = pSrcPicture->pDrawable;
-    int w = pDraw->width;
-    int h = pDraw->height;
+
+    if (!pDraw)
+        return FALSE;
+
+    w = pDraw->width;
+    h = pDraw->height;
 
     if (op > PictOpAdd)
 	return FALSE;
